### DISCUSSION TITLE
Insert Link - but only the Zettelkasten ID part

### DISCUSSION THREAD

**Discussion by ibnishak** [nvim-telekasten/telekasten.nvim#170] [Ideas]
I guess it could be add an a parameter to the insert_link subcommand.

**Comment by lambtho12** [comment]
Could you provide a more detailed explanation of what you want to achieve ?

I assume you have your notes named following `uuid-title`. Inserting a link in the form of `[[uuid]]` will not create a link to the actual file. Then trying to follow this link will make telekasten prompt you to create the note called `uuid.md` instead. Is it what you want ?

**Reply by ibnishak** [reply]
Following link to `[[UUID]]` will bring up telescope picker which will show exisitng files with that UUID. This enables updating filenames without having to update links - which is tad risky when automated.

**Reply by lambtho12** [reply]
Ok. This makes sense. I am quite busy currently, so I do not know when I will have some time to implement that (maybe next week?), but I would be happy to review a PR is you make one in the meantime.

A good idea would be to make a small helper function like: `extract_uuid(filename, uuid_type)` so we can reuse that in other places as well  (to fix https://github.com/renerocksai/telekasten.nvim/issues/138 for instance)
